libs/basics/src/iterators/async_iterator_plus.ts

```diff
@@ -294,6 +294,33 @@ export class AsyncIteratorPlusImpl<T> implements AsyncIteratorPlus<T> {
     return set;
   }
 
+  windows(groupSize: 0): never;
+  windows(groupSize: 1): AsyncIteratorPlus<[T]>;
+  windows(groupSize: 2): AsyncIteratorPlus<[T, T]>;
+  windows(groupSize: 3): AsyncIteratorPlus<[T, T, T]>;
+  windows(groupSize: 4): AsyncIteratorPlus<[T, T, T, T]>;
+  windows(groupSize: 5): AsyncIteratorPlus<[T, T, T, T, T]>;
+  windows(groupSize: number): AsyncIteratorPlus<T[]>;
+  windows(groupSize: number): AsyncIteratorPlus<T[]> {
```

Comment:
Wouldn't it be nice if there was a better way to do this? I would guess other languages have this right? But it would be relying on a runtime-variable to set the compile-time length of an array, so I'm not sure if it is possible...

Reply:
Yeah, I tried to make this happen using type magic but TS doesn't like it because it has a hard time knowing how deep to go:
<img width="1645" alt="image" src="https://user-images.githubusercontent.com/1938/227028467-07f4dd9f-7c8d-483a-82ae-010b8b5097cd.png">